opensearchtransport/opensearchtransport.go

```diff
@@ -407,14 +402,20 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 			time.Sleep(c.retryBackoff(i + 1))
 		}
 	}
+	// Read, close and replace the http reponse body to close the connection
+	if res != nil && res.Body != nil {
+		body, err := io.ReadAll(res.Body)
+		res.Body.Close()
```

Comment:
@Jakob3xD Just a quick question: since we are closing the response body here, would it cause errors for the user if they try to close it again in their application?

Reply:
See above, "this wont brake any existing implementations as you can still call .Close() on the Body.". Still needs a unit test.

Reply:
To explain it in addition. The NopeCluster returns a io.ReadCloser like the res.Body is and has a Close() func that just returns nil. https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/io/io.go;l=678
So you can still call close, but it won't do anything as it is not needed.